src/Kernel/Magic/ViewMagic.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Jupyter.Core.Protocol;
+using Microsoft.Quantum.IQSharp.Core.ExecutionPathTracer;
+using Microsoft.Quantum.IQSharp.Jupyter;
+using Microsoft.Quantum.Simulation.Simulators;
+using Newtonsoft.Json;
+
+namespace Microsoft.Quantum.IQSharp.Kernel
+{
+    /// <summary>
+    ///      Contains the JSON representation of the <see cref="ExecutionPath"/>
+    ///      and the ID of the HTML div that will contain the visualization.
+    /// </summary>
+    public class ExecutionPathVisualizerContent : MessageContent
+    {
+        /// <summary>
+        ///     Initializes <see cref="ExecutionPathVisualizerContent"/> with the
+        ///     given JSON string and HTML div ID.
+        /// </summary>
+        public ExecutionPathVisualizerContent(string json, string id)
+        {
+            this.Json = json;
+            this.Id = id;
+        }
+
+        /// <summary>
+        ///     JSON representation of the <see cref="ExecutionPath"/>.
+        /// </summary>
+        [JsonProperty("json")]
+        public string Json { get; }
+
+        /// <summary>
+        ///     ID of the HTML div that will contain the visualization.
+        /// </summary>
+        [JsonProperty("id")]
+        public string Id { get; }
+    }
+
+    /// <summary>
+    ///     A magic command that can be used to visualize the execution
+    ///     paths of operations and functions.
+    /// </summary>
+    public class ViewMagic : AbstractMagic
+    {
+        private const string ParameterNameOperationName = "__operationName__";
+
+        /// <summary>
+        ///     Constructs a new magic command given a resolver used to find
+        ///     operations and functions, and a configuration source used to set
+        ///     configuration options.
+        /// </summary>
+        public ViewMagic(ISymbolResolver resolver, IConfigurationSource configurationSource) : base(
```

Comment:
The name `%view` is a bit generic; what is it that I'm viewing? It may be good to make it slightly more verbose, such as `%view-trace`?

Reply:
Would `%visualize` make more sense? It conveys more of the idea of visualizing a Q# operation whereas `trace` might not be as obvious to the end user.

Reply:
We could also open this up on our internal channel to get feedback from others regarding the best name for the magic command?

Reply:
cc-ing @anpaz-msft and @rmshaffer for visibility

Reply:
In my mind, `%trace` is more clearly associated with a program execution path than `%visualize`, which has a bit of the same problem as `%view` in being a bit generic. I agree, more brainstorming would be good here.

Reply:
For context, there's also issue https://github.com/microsoft/iqsharp/issues/25 which is suggesting a command called `%printcode`. Maybe we come up with some consistent naming scheme that could be used for both scenarios. e.g. `%viewtrace` and `%viewcode`, or perhaps `%printtrace` and `%printcode`.

Reply:
`#viewtrace` makes more sense to me than `%printtrace` because `print` gives me the impression that it is text-based rather than an interactive HTML element. On the other hand, `%printcode` might make more sense since it's text-based. 
Then again, I'm not the best at naming, so I'll defer the decision to you guys haha!

Reply:
Agreed, I didn't like `%printtrace` from the moment I typed it. 😊 Personally I like the simplicity of just `%trace`, since the word `view` doesn't seem like it adds much value, but I'd be fine either way.

Reply:
Yea, I think `%trace` sounds like a good option! @anpaz-msft @cgranade: thoughts on `%trace` instead of `%view`?

Reply:
I like `%trace` in that it helps reinforce that the output isn't a visualization of the program, but of a particular path through that program. We don't in general expect tracing and debugging tools, for instance, to show every possible path through a program; similarly, we shouldn't expect to be able to reconstruct a Q# operation from a visualization of its trace.